core/domain/collection_domain_test.py

```diff
@@ -318,30 +329,64 @@ def test_add_delete_node(self):
         collection.delete_node('test_exp')
         self.assertEqual(len(collection.nodes), 0)
 
-    def test_skills_property(self):
+    def test_add_skill(self):
```

Comment:
Although existing tests don't do this, well formed test names should include both the condition being tested and the expected outcome. If multiple, different conditions are being tested in the same test method, they should be split across multiple test methods.